lib/periodicChecks.js

```diff
@@ -0,0 +1,234 @@
+    context.repo({
+      issue_number: pullRequest.number,
+      assignees: [pullRequest.user.login],
+    })
+  );
+  await context.github.issues.createComment(
+    context.repo({
+      issue_number: pullRequest.number,
+      body: comment,
+    })
+  );
+};
+
+/**
+ * This function handles the state when a pull request has been approved.
+ * 1. Check if the PR author has merging rights and assign them to the PR.
+ * 2. If the PR author does not have merging rights, assign the PR to the
+ *    project owner from the changelog label.
+ * 3. If the pull request does not have a changelog label, assign the pr
+ *    author asking them to ping one of the reviewers to merge the PR.
+ *
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const handleApprovedPR = async (context, pullRequest) => {
+  // Check if author can merge PR. All members can merge the pull request.
+  const authorCanMerge = await isUserAMemberOfTheOrganisation(
+    context,
+    pullRequest.user.login
+  );
+
+  if (authorCanMerge) {
+    // Assign author.
+    await pingAndAssignPrAuthor(
+      context,
+      pullRequest,
+      'Hi @' +
+        pullRequest.user.login +
+        ', this PR is ready to be merged. Please make sure there are no ' +
+        'pending comments before merge. Thanks!'
+    );
+  } else {
+    // Assign project owner to merge the pull request.
+    // A pull request will always have a changelog label because we are
+    // assigning the PR author to the pull request on creation if created
+    // without a changelog label.
+    const label = getChangelogLabelFromPullRequest(pullRequest);
+    const projectOwner = getProjectOwnerFromLabel(label);
+    await context.github.issues.addAssignees(
+      context.repo({
+        issue_number: pullRequest.number,
+        assignees: [projectOwner],
+      })
+    );
+    await context.github.issues.createComment(
+      context.repo({
+        issue_number: pullRequest.number,
+        body:
+          'Hi @' +
+          projectOwner +
+          ', this PR is ready to be merged. Please make sure there are no ' +
+          "pending comments from the author's end before merge. Thanks!",
+      })
+    );
+
+  }
+};
+
+/**
+ * This function ensures that a pull request is assigned.
+ * 1. Fetch all pull requests.
+ * 2. Filter out those without assignees.
+ * 3. Find out appropriate assignee.
+ *  - If pull request has a merge conflict    -- PR author.
+ *  - If pull request has pending reviews     -- Remaining reviewers.
+ *  - If pull request has changes requested   -- PR author.
+ *  - If pull request has been approved       -- Any member of the Organisation.
+ *  - If none of the above cases              -- Welfare team lead.
+ * 4. Assign appropriate assignee.
+ * 5. Ping appropriate assignee.
+ * @param {import('probot').Context} context
+ */
+const ensurePullRequestIsAssigned = async (context) => {
```

Comment:
So, one high level comment I have in mind - why not directly use the `checkPullRequestReview` module?  We are doing the same things right, just on a periodic basis?

Reply:
It would make the code simpler and cleaner as well.

Reply:
Not everything here is used in the `checkPullRequestReview` module. I've modified the code to use some functions from the `checkPullRequestReview` module.